lib/tasks/active_record_views.rake

```diff
@@ -52,12 +52,46 @@ Rake::Task[schema_rake_task].enhance do
     pg_tasks = tasks.send(:class_for_adapter, adapter).new(config)
     pg_tasks.send(:set_psql_env)
 
-    require 'shellwords'
-    system("pg_dump --data-only --table=active_record_views #{Shellwords.escape database} >> #{Shellwords.escape filename}")
-    raise 'active_record_views metadata dump failed' unless $?.success?
+    begin
+      active_record_views_dump = Tempfile.open("active_record_views_dump.sql")
+      require 'shellwords'
+      system("pg_dump --data-only --no-owner --table=active_record_views #{Shellwords.escape database} >> #{Shellwords.escape active_record_views_dump.path}")
+      raise 'active_record_views metadata dump failed' unless $?.success?
 
-    File.open filename, 'a' do |io|
-      io.puts 'UPDATE public.active_record_views SET refreshed_at = NULL WHERE refreshed_at IS NOT NULL;'
+      ActiveRecordViews::RakeTaskUtilities.remove_sql_header_comments(active_record_views_dump.path)
```

Comment:
We already rely on internal methods in `ActiveRecord::Tasks::DatabaseTasks` for other things in this code path so I’m tempted to do the same here with something like:

```ruby
if Gem::Version.new(Rails.version) >= Gem::Version.new("5.1")
  pg_tasks.send(:remove_sql_header_comments, active_record_views_dump.path)
end
```

Since the cleaning of `structure.sql` was only introduced in Rails 5.1, I think it’s more than fine for ARV to only do the same on Rails 5.1+.

This will save us from having to duplicate code from Rails and will mean we get any future enhancements to this method “for free”. (I bump the gemspec for each Rails version due to the general tight coupling this gem has with ActiveRecord/Rails so I’m not too worried about unexpected breakages.)

Reply:
That sounds good to me, gets rid of my ugly module name too :)